server/src/streaming/systems/system.rs

```diff
@@ -57,11 +63,12 @@ pub struct System {
     pub(crate) storage: Arc<SystemStorage>,
     pub(crate) streams: HashMap<u32, Stream>,
     pub(crate) streams_ids: HashMap<String, u32>,
+    pub(crate) users: HashMap<UserId, User>,
     pub(crate) config: Arc<SystemConfig>,
     pub(crate) client_manager: IggySharedMut<ClientManager>,
     pub(crate) encryptor: Option<Box<dyn Encryptor>>,
     pub(crate) metrics: Metrics,
-    pub(crate) db: Option<Arc<Db>>,
+    pub(crate) state: Arc<dyn State>,
```

Comment:
Any idea how we would get rid of this dynamic dispatch for the great unification with `io_uring` branch ? maybe having this as generic parameter on the `IggyShard` from `io_uring` branch or System from master ?

Reply:
Didn't think of it yet, but this is certainly worth considering.